kasa/smart/modules/lighttransition.py

```diff
@@ -67,54 +88,90 @@ def _create_features(self):
                 )
-            )  # self._turn_off_transition_max
-
-    @property
-    def _turn_on(self):
-        """Internal getter for turn on settings."""
-        if "on_state" not in self.data:
-            raise KasaException(
-                f"Unsupported for {self.REQUIRED_COMPONENT} v{self.supported_version}"
             )
 
-        return self.data["on_state"]
-
-    @property
-    def _turn_off(self):
-        """Internal getter for turn off settings."""
-        if "off_state" not in self.data:
+    def _post_update_hook(self) -> None:
+        """Update the states."""
+        # Assumes any device with state in sysinfo supports on and off and
+        # has maximum values for both.
+        # v2 adds separate on & off states
+        # v3 adds max_duration except for ks240 which is v2 but supports it
+        if not self._supports_on_and_off:
+            self._enabled = self.data["enable"]
+            return
+
+        if self._state_in_sysinfo:
+            on_max = self._device.sys_info.get(
+                "max_fade_on_time", self.MAXIMUM_DURATION
+            )
+            off_max = self._device.sys_info.get(
+                "max_fade_off_time", self.MAXIMUM_DURATION
+            )
+            on_enabled = bool(self._device.sys_info["gradually_on_mode"])
+            off_enabled = bool(self._device.sys_info["gradually_off_mode"])
+            on_duration = self._device.sys_info["fade_on_time"]
+            off_duration = self._device.sys_info["fade_off_time"]
+        elif (on_state := self.data.get("on_state")) and (
+            off_state := self.data.get("off_state")
+        ):
+            on_max = on_state.get("max_duration", self.MAXIMUM_DURATION)
+            off_max = off_state.get("max_duration", self.MAXIMUM_DURATION)
+            on_enabled = on_state["enable"]
+            off_enabled = off_state["enable"]
+            on_duration = on_state["duration"]
+            off_duration = off_state["duration"]
+        else:
             raise KasaException(
                 f"Unsupported for {self.REQUIRED_COMPONENT} v{self.supported_version}"
             )
 
-        return self.data["off_state"]
-
-    async def set_enabled_v1(self, enable: bool):
+        self._enabled = on_enabled or off_enabled
+        self._on_state = {
+            "duration": on_duration,
+            "enable": on_enabled,
+            "max_duration": on_max,
+        }
+        self._off_state = {
+            "duration": off_duration,
+            "enable": off_enabled,
+            "max_duration": off_max,
+        }
+
+    async def set_enabled(self, enable: bool):
         """Enable gradual on/off."""
-        return await self.call("set_on_off_gradually_info", {"enable": enable})
+        if not self._supports_on_and_off:
+            return await self.call("set_on_off_gradually_info", {"enable": enable})
+        else:
+            duration = self.DEFAULT_DURATION if enable else 0
```

Comment:
I don't think we should reset the existing setting to default when enabling on v2+.

Reply:
Ok that's fixed